regression_tests/utils.go

```diff
@@ -0,0 +1,222 @@
+package regressiontests
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"regexp"
+	"strings"
+	"time"
+
+	walletclient "github.com/bitcoin-sv/spv-wallet-go-client"
+	"github.com/bitcoin-sv/spv-wallet-go-client/xpriv"
+	"github.com/bitcoin-sv/spv-wallet/models"
+	"github.com/bitcoin-sv/spv-wallet/models/filter"
+)
+
+const (
+	atSign                   = "@"
+	domainPrefix             = "https://"
+	domainSuffixSharedConfig = "/v1/shared-config"
+
+	ClientOneURLEnvVar         = "CLIENT_ONE_URL"
+	ClientTwoURLEnvVar         = "CLIENT_TWO_URL"
+	ClientOneLeaderXPrivEnvVar = "CLIENT_ONE_LEADER_XPRIV"
+	ClientTwoLeaderXPrivEnvVar = "CLIENT_TWO_LEADER_XPRIV"
+
+	timeoutDuration = 30 * time.Second
+)
+
+var (
+	explicitHTTPURLRegex = regexp.MustCompile(`^https?://`)
+	envVariableError     = errors.New("missing xpriv variables")
+)
+
+type regressionTestUser struct {
+	XPriv   string `json:"xpriv"`
+	XPub    string `json:"xpub"`
+	Paymail string `json:"paymail"`
+}
+
+type regressionTestConfig struct {
+	ClientOneURL         string
+	ClientTwoURL         string
+	ClientOneLeaderXPriv string
+	ClientTwoLeaderXPriv string
+}
+
+// getEnvVariables retrieves the environment variables needed for the regression tests.
+func getEnvVariables() (*regressionTestConfig, error) {
+	rtConfig := regressionTestConfig{
+		ClientOneURL:         os.Getenv(ClientOneURLEnvVar),
```

Comment:
why not use units test env vars ? `t.Setenv()` u just would need to pass t testing to `getEnvVariables()`